### PR TITLE
Rspec tests

### PR DESCRIPTION
**Pull request by Dgrafmyre**
Add test gems
Add rspec test directory 
Configure RSpec settings for test gems
Create Comments feature test for all three comment forms

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,6 +18,7 @@
 client-bundle.js
 client-bundle.js.map
 .env
+.idea/
```
Comment (justin808) [inline .gitignore]: This should be in your global git ignore, not the project one.

File: config/environments/production.rb [hunk 2]
```diff
@@ -20,7 +20,7 @@
   # config.action_dispatch.rack_cache = true
 
   # Disable Rails's static asset server (Apache or NGINX will already do this).
-  config.serve_static_assets = false
+  config.serve_static_files = false
```
Comment (justin808) [inline config/environments/production.rb]: OK -- fixes deprecation message.

**Comment by justin808** [comment]
Looks good. Ready for merge?

**Comment by Dgrafmyre** [comment]
I removed .idea from project .gitignore and added it to global. Should be good to go! Thanks!

**Comment by Dgrafmyre** [comment]
At this time Capybara-Webkit and Poltergeist do not work as headless drivers with Capybara
 and ReactJS without custom javascript implementation and modified css.
 https://github.com/thoughtbot/capybara-webkit/issues/494
 https://github.com/teampoltergeist/poltergeist/issues/292
The possible workout arounds create test code that is not easy to understand with required customization and might become brittle in future development.

**Comment by Dgrafmyre** [comment]
By adding node module ES5-SHIM(https://github.com/es-shims/es5-shim) and updating from qt to qt5(https://github.com/thoughtbot/capybara-webkit/issues/494) on my mac, I was able to use Capybara-Webkit instead of Selenium-Webdriver. This enabled faster, headless RSpec feature tests.

**Pull request merged**